metals/src/main/scala/scala/meta/internal/metals/JarTopLevels.scala

```diff
@@ -29,10 +27,10 @@ final class JarTopLevels(conn: () => Connection) {
    */
   def getTopLevels(
       path: AbsolutePath
-  ): Option[TrieMap[String, AbsolutePath]] =
+  ): Option[List[(String, AbsolutePath)]] =
     try {
       val fs = PlatformFileIO.newJarFileSystem(path, create = false)
-      val toplevels = TrieMap[String, AbsolutePath]()
+      val toplevels = List.newBuilder[(String, AbsolutePath)]
```

Comment:
Out of curiosity, why the change from the `TrieMap` to `List`? Normally I prefer to have a Map for this over a list of tuples unless there is a good reason not to.

Reply:
`scala.collection.concurrent.TrieMap` is too complex type for this method. It was used because `OnDemandSymbolIndex.toplevels` had this signature and there was this tricky method: https://github.com/scalameta/metals/blob/03704c17b5397bd8feb4771d0bedf9a6dfe0a660/mtags/src/main/scala-2/scala/meta/internal/mtags/OnDemandSymbolIndex.scala#L96-L104

But as `toplevels` signature was changed to `TrieMap[String, Set[AbsolutePath]]` it became inconvenient to use `addSourceJarTopLevels`  as they were so I changed this part and replaced `TrieMap` here on smth more suitable.

It might be `Map` but it anyway used only as a `List`.

Reply:
Alright, sounds good and thanks for the explanation.